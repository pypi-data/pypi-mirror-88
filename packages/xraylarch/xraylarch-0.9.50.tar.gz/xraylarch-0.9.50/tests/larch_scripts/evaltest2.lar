x = 3
a = arange(7)
b = arange(64)
b.shape = (8,8)
c = arange(10.)/10
u = 0.5
x = 3
y = 2
z = 7.4
alist = [[1.,2.,3.,4.],[6.,7.,8.,9.],['a','b','c','d']]
adict ={'a':1.0, 'b':2.0, 'c':3.0}
yes = True
no  = False
nottrue = False
orx =2
andy= 99

def f1(a,b,c=10):
   print 'f1'
    return (a + b)/ c
#enddef

def f2(c=100,**kw):
    return c/2.0
#enddef
def f3(x,b):
    return x * b
#enddef

## begin

f3(2.0,5)                
a[2] + 3                 
b[2][4]                  
a[3+1]                   

x == 2                   
x == 3                   
x != 3                   
x >= 1                   
x >= 3                   
x >= 4                   
x < 99                   
(x>u) or x*u>100         
yes and no or nottrue     
yes and (no or nottrue)   
(yes and no) or nottrue   
yes or no and nottrue     
yes or (no and nottrue)   
(yes or no) and nottrue   
yes or not no             
not (no or yes)           
not no or yes             
not yes                   
not no                    
not yes                     
not no                      
a[0]                      
a[1]                      
a[0] and a[1]             
not a[0]                  
not a[1]                  
not a[2]                  

f1(2,8.0)                 
f1(2,2*7.,c=11.0)         
f1(2,8.0,c=sin(0.3))      
f2()                      
f2(c=99)                  
f2(d=88)                  
max(a[3]*01,9,3)          
sqrt(33.2j / cos(88.0))  
adict['b']                

adict['c'] /adict['b']    
0.2                       
.2                        
0.223e+4                  
.32101e+3j                
 
adict                     


[1,2,['a', 'b','c'], 4.0] 
alist[2]                  
{'a':1, 'b':2.0, 'c':[1,2,'three']} 
alist                     


-(a[3:5] + 2)             
-a[3:7] + 2.0             

a[2:]                    
'a simple string '        
alist[2][3]               
[0,1,2,4,8,16,32]         
alist[1]                  
f3(2.5,b)[3,4]            
 
' %s = %g '  % ['sqrt(22.3)',sqrt(22.3)]   

b[2:4,3]                   
b[2:4,3:5]                 
b[2:4]                     
f3(2.5,b)[3]               
sin(a*pi/20)               




