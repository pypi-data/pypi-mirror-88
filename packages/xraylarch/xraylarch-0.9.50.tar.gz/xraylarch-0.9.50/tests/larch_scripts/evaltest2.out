10.0
5
20
4
False
True
False
True
True
False
True
True
False
False
False
True
True
False
True
False
True
False
True
False
True
0
1
0
True
False
False
1.0
1.45454545455
33.8386336182
50.0
49.5
50.0
9.0
(4.0755870761,4.0755870761j)
2
1.5
0.2
.2
0.223e+4
(0,.32101e+3j)
{'a':1.0,'c':3.0,'b':2.0}
[1.0,2.0,['a','b','c'],4.0]
['a','b','c','d']
{'a':1.0,'c':[1.0,2.0,'three'],'b':2.0}
[[1.0,2.0,3.0,4.0],[6.0,7.0,8.0,9.0],['a','b','c','d']]
[-5.-6.]
[-1.-2.-3.-4.]
[2.3.4.5.6.]
'asimplestring'
'd'
[0.1.2.4.8.16.32.]
[6.7.8.9.]
70.0
'sqrt(22.3)=4.72229'
[1927]
[[1920][2728]]
[[1617181920212223][2425262728293031]]
[60.62.565.67.570.72.575.77.5]
[0.0.156434470.309016990.45399050.587785250.70710678
0.80901699]
